### ISSUE TITLE
Bug in $action-setfield: Deletes 'text' field by default

### ISSUE THREAD

**Issue by Drakor** [Jermolene/TiddlyWiki5#1968]
Hi,

#1963 introduced a bug into the $action-setfield that causes it to delete the 'text' field of a tiddler by default, when using it without the '$' parameters. Example:
```
<$action-setfield $tiddler="testResults" tag="myTag" myField="test" />
```

A potential fix is here #1967, an example wiki that showcases both the fix and the bug can be found here: http://twguides.org/contrib/fixActionSetField.html

/Andreas

**Comment by tobibeer** [comment]
Thanks for catching this before release.

Rather than #1967. I think it might be safer to fix **wiki.js**, so that `setText()` checks if the `field` parameter is actually set, before setting the default being field `text` to **undefined**.

So, this [else](https://github.com/Jermolene/TiddlyWiki5/blob/master/core/modules/wiki.js#L74) perhaps better be...

`else if (typeof field == "string" || typeof value == "string") {`

In other words, if you wanted to literally reset the `text` field you have to at least specify that that's the field you want to reset.

**Comment by Drakor** [comment]
Yes, that is another option. Also most mentions of setText or setTextReference elsewhere in the core seem unaffected with the exception that

```
<$button set="myTiddler">
Click me
</$button>
```

will no longer clear the tiddler (with setTo="" it will still work, since "" is a string and that is also the far more likely use from someone who wants to clear the tiddlers text). Can you make a quick pull request for that ?

**Comment by tobibeer** [comment]
I'd wait for @Jermolene's feedback and then you will or I will, or he already did it. You sure noticed this bug that I wouldn't have thought of and delivered a way to smash it. Personally, I don't care so much who's doing any PRs, so long as bugs are shown the door and things eventually work as expected.

**Comment by tobibeer** [comment]
@Jermolene, I think fixing this is important for 5.1.10.

**Comment by Jermolene** [comment]
I'm inclined to go with @Drakor's proposal so that we don't inadvertently break code depending on the current semantics of `wiki.setText()`

**Comment by Jermolene** [comment]
Closed by #1967
